hivemind/dht/validation.py

```diff
@@ -52,3 +53,71 @@ def strip_value(self, record: DHTRecord) -> bytes:
         """
 
         return record.value
+
+    @property
+    def priority(self) -> int:
+        """
+        Defines the order of applying this validator with respect to other validators.
+
+        The validators are applied:
+          - In order of increasing priority for signing a record
+          - In order of decreasing priority for validating and stripping a record
+        """
+
+        return 0
+
+    def merge_with(self, other: 'RecordValidatorBase') -> bool:
+        """
+        By default, all validators are applied sequentially (i.e. we require all validate() calls
+        to return True for a record to be validated successfully).
+
+        However, you may want to define another policy for combining your validator classes
+        (e.g. for schema validators, we want to require only one validate() call to return True
+        because each validator bears a part of the schema).
+
+        This can be achieved with overriding merge_with(). It should:
+
+          - Return True if it has successfully merged the `other` validator to `self`,
+            so that `self` became a validator that combines the old `self` and `other` using
+            the necessary policy. In this case, `other` should remain unchanged.
+
+          - Return False if the merging has not happened. In this case, both `self` and `other`
+            should remain unchanged. The DHT will try merging `other` to another validator or
+            add it as a separate validator (to be applied sequentially).
+        """
+
+        return False
+
+
+class CompositeValidator(RecordValidatorBase):
+    def __init__(self, validators: List[RecordValidatorBase]=None):
+        self._validators = []
+        if validators is not None:
+            self.extend(validators)
+
+    def extend(self, validators: List[RecordValidatorBase]) -> None:
```

Comment:
```suggestion
    def extend(self, validators: List[RecordValidatorBase]):
```

Reply:
Why is that?

I thought the [convention](https://github.com/python/typing/issues/65#issuecomment-96904050) is that this means no annotation, and to add one we should explicitly specify `-> None` (like we do in other part of the project, e.g. [here](https://github.com/learning-at-home/hivemind/blob/master/hivemind/dht/__init__.py#L65)).